packages/l/libsdl/xmake.lua

```diff
@@ -192,6 +196,9 @@ package("libsdl")
                 end
                 table.insert(configs, "-DCMAKE_INCLUDE_PATH=" .. table.concat(includedirs, ";"))
             end
+        if package:config("with_wayland") then
+            opt.envs = {PKG_CONFIG_PATH = package:dep("wayland"):installdir("lib", "pkgconfig")}
```

Comment:
you need only add `add_deps("wayland")`, it will add PKG_CONFIG_PATH automatically.

Reply:
In that case, without setting manually PKG_CONFIG_PATH, it does not detect wayland.
The thing is, sdl compiles fine without it, so it's hard to know whether or not it compiled with wayland.
I just tried to install libsdl again (with the official repo) so it downloaded wayland as expected but I got during the cmake process (of sdl) :
```
-- Checking for modules 'wayland-client>=1.18;wayland-egl;wayland-cursor;egl;xkbcommon>=0.5.0'
--   No package 'wayland-client' found
--   No package 'wayland-egl' found
--   No package 'wayland-cursor' found
``` 
and a little after
```
--   SDL_WAYLAND                 (Wanted: ON): OFF
--   SDL_WAYLAND_LIBDECOR        (Wanted: ON): OFF
--   SDL_WAYLAND_LIBDECOR_SHARED (Wanted: ON): OFF
--   SDL_WAYLAND_QT_TOUCH        (Wanted: ON): OFF
--   SDL_WAYLAND_SHARED          (Wanted: ON): OFF
```
even though sdl was built just fine
I have no idea on how, but I should try to add a test if sdl compiled with wayland

Reply:
try this patch https://github.com/xmake-io/xmake/pull/4165

```
xmake update -s github:xmake-io/xmake#cmake
```

Reply:
I just did
```
xmake update -s github:xmake-io/xmake#cmake
xmake require -v -j 6 --extra={system=false} libsdl
```
and it still does not work

Reply:
It should work, you can print buildenvs and check PKG_CONFIG_PATH

```
package("libsdl")
    on_install()
        print(import("package.tools.cmake").buildenvs(package))
    end
```

Reply:
try it again. 

```
xmake update -s github:xmake-io/xmake#cmake
```